chessboard/skoved/chessboard.go

```diff
@@ -0,0 +1,59 @@
+package chessboard
+
+// Declare a type named Rank which stores if a square is occupied by a piece - this will be a slice of bools
+type Rank []bool
+
+// Declare a type named Chessboard contains a map of eight Ranks, accessed with values from "A" to "H"
+type Chessboard map[string]Rank
+
+// validateFile checks if an int is a valid chessboard file (1-8)
+func validateFile(file int) bool {
+	return file >= 1 && file <= 8
+}
+
+// CountInRank returns how many squares are occupied in the chessboard,
+// within the given rank
+func CountInRank(cb Chessboard, rank string) int {
+	numOccupied := 0
+	row, exists := cb[rank]
+	if exists {
```

Comment:
Great man! I think you are the only one that has done this check. I'll add it to my solution.